src/AnonymousUtils.js

```diff
@@ -0,0 +1,113 @@
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow-weak
+ */
+import ParseUser from './ParseUser';
+const uuidv4 = require('uuid/v4');
+
+let registered = false;
+
+const authenticationProvider = {
+  authData: {
+    authData: {
+      id: uuidv4(),
+    }
+  },
+
+  restoreAuthentication() {
+    return true;
+  },
+
+  getAuthType() {
+    return 'anonymous';
+  },
+};
+
+/**
+ * Provides utility functions for working with Anonymously logged-in users. <br />
+ * Anonymous users have some unique characteristics:
+ * <ul>
+ *  <li>Anonymous users don't need a user name or password.</li>
+ *  <ul>
+ *    <li>Once logged out, an anonymous user cannot be recovered.</li>
+ *  </ul>
+ *  <li>signUp converts an anonymous user to a standard user with the given username and password.</li>
+ *  <ul>
+ *    <li>Data associated with the anonymous user is retained.</li>
+ *  </ul>
+ *  <li>logIn switches users without converting the anonymous user.</li>
+ *  <ul>
+ *    <li>Data associated with the anonymous user will be lost.</li>
+ *  </ul>
+ *  <li>Service logIn (e.g. Facebook, Twitter) will attempt to convert
+ *  the anonymous user into a standard user by linking it to the service.</li>
+ *  <ul>
+ *    <li>If a user already exists that is linked to the service, it will instead switch to the existing user.</li>
+ *  </ul>
+ *  <li>Service linking (e.g. Facebook, Twitter) will convert the anonymous user
+ *  into a standard user by linking it to the service.</li>
+ * </ul>
+ * @class Parse.AnonymousUtils
+ * @static
+ */
+const AnonymousUtils = {
+  /**
+   * Gets whether the user has their account linked to anonymous user.
+   *
+   * @method isLinked
+   * @name Parse.AnonymousUtils.isLinked
+   * @param {Parse.User} user User to check for.
+   *     The user must be logged in on this device.
+   * @return {Boolean} <code>true</code> if the user has their account
+   *     linked to an anonymous user.
+   * @static
+   */
+  isLinked(user) {
+    const provider = this._getAuthProvider();
+    return user._isLinked(provider.getAuthType());
+  },
+
+  /**
+   * Logs in a user Anonymously.
+   *
+   * @method logIn
+   * @name Parse.AnonymousUtils.logIn
+   * @returns {Promise}
+   * @static
+   */
+  logIn() {
+    const provider = this._getAuthProvider();
+    return ParseUser._logInWith(provider.getAuthType(), provider.authData);
+  },
+
+  /**
+   * Links Anonymous User to an existing PFUser.
+   *
+   * @method link
+   * @name Parse.AnonymousUtils.link
+   * @param {Parse.User} user User to link. This must be the current user.
+   * @returns {Promise}
+   * @static
+   */
+  link(user) {
+    const provider = this._getAuthProvider();
+    return user._linkWith(provider.getAuthType(), provider.authData);
+  },
+
+  _getAuthProvider() {
+    const provider = authenticationProvider;
```

Comment:
this is a no-op. You could just take out this line and change 106 to:

```
ParseUser._registerAuthenticationProvider(authenticationProvider);
```

but even better is moving the declaration of `authenticationProvider` into this function so that the `uuid` is only generated IF it is needed.  As you have it here, any time this file is required, it'll generate the `uuid` which is relatively expensive.  Make it lazy so it is only generated if needed.